### PR TITLE
Shared CI configurations

### PR DESCRIPTION
**Pull request by leobr29**
Shared CI configuration to enable consistency, avoid duplication and code reusability

### CODE REVIEW

**Comment by leobr29** [comment]
Clickup https://app.clickup.com/t/8696mhjmt

File: .github/workflows/component-ts-test.yml [hunk 1]
```diff
@@ -16,57 +16,40 @@ jobs:
...
-        uses: actions/upload-artifact@v4
-        with:
-          name: jest-test-results
-          path: ./typescript-demo/test-results/junit.xml
-
-      - name: Display test results in summary
-        uses: dorny/test-reporter@v1
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v5
+        if: success() || failure() # always run even if the previous step fails
```
Comment (roje-bodc) [inline .github/workflows/component-ts-test.yml:45]: Again, could this be always() instead?

File: .github/workflows/workflow-ts.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    branches:
+      # Run on our main branch
+      - main
+    paths:
+      - typescript-demo/**
+  pull_request:
+    # Run for any pull requests
+    paths:
+      - typescript-demo/**
+  # TODO : Do we have to keep that ?!
```
[the remark this answers is not in the document]
Comment (leobr29) [reply .github/workflows/workflow-ts.yml:14]: Good point !

**Comment by leobr29** [comment]
@roje-bodc I made the changes

**Review by roje-bodc — APPROVED** [review]
All looks good to me 👍

**Pull request merged**